server/asset/common.go

```diff
@@ -92,6 +95,9 @@ type OutputTracker interface {
 type AccountBalancer interface {
 	// AccountBalance retrieves the current account balance.
 	AccountBalance(addr string) (uint64, error)
+	// ValidateSignature checks that the pubkey is correct for the address and
+	// that the signature shows ownership of the associated private key.
+	ValidateSignature(addr string, pubkey, msg, sig []byte) error
```

Comment:
Thinking of just updating the docs here to direct enforcement of an address standard.

```
// ValidateSignature checks that the pubkey is correct for the address and
// that the signature shows ownership of the associated private key. The
// address string will be validated against a strict standard format. For
// example, ethereum address should have the "0x" prefix and be lower-case.
// Any other format should return an error.
```

Reply:
`ValidateSignature`, being an ETH backend method, shouldn't have trouble with any of the normal string encodings.

`common.HexToAddress` handles all valid variants regardless of case or 0x prefix.
https://pkg.go.dev/github.com/ethereum/go-ethereum/common#HexToAddress

Reply:
But this string encoding is used as a unique ID throughout dcrdex.

Reply:
Oh, it certainly needs to be standardized for all the other uses, yes, I just don't think `ValidateSignature` is the place where it matters and hence not the best place to doc the standard we pick.

Reply:
`ValidateSignature` is run on all incoming orders. It seems like a great place to validate the signature, no?

Reply:
Yes, to validate the signature, just not to document the expected address format IMO.

Reply:
I'm not sure we're on the same page. I'm trying to instruct asset backend developers to validate the address string against a strict stadard. I'm just using ethereum as an example.

Reply:
This is under the assumption that we use the scheme defined in this PR, which is that the address is either the `msgjson.Trade.Address` or `string(msgjson.Trade.Coins[0])`.

Reply:
Alright, well since order router and ValidateSignature are the initial validation points for the address string, I suppose have it be strict about the encoding, more strict than it itself would require?

Reply:
That's what I'm trying to get at. Can improve the language, it seems.

Reply:
Another approach would be for `ValidateSignature` to return a canonicalized address string after processing whatever string came in via the `msgjson.Trade` fields.

The fact that `msgjson.Trade.Address` has always been a string and that's hard to change now kind means we have to have stick with a string for the other address, but @davecgh did have a nice suggestion if we had bytes that could be put in a type like `type Address interface { fmt.Stringer; Bytes() []byte }`, but alas the issue is upstream of that with the communications type committing us to a string.  So I think we have to work with it as-is.